File: android/app/src/main/java/com/pan3d/material/MaterialLoad.java
package com.pan3d.material;

import com.pan3d.program.Shader3D;

import java.util.HashMap;

public class MaterialLoad {

    public MaterialBackFun fun;
    public HashMap info;
    public String url;
    public boolean autoReg;
    public String regName;
    public Shader3D shader3D;

    public MaterialLoad(MaterialBackFun $fun, HashMap $info, String $url, boolean $autoReg, String $regName, Shader3D $shader3D) {
        this.fun = $fun;
        this.info = $info;
        this.url = $url;
        this.autoReg = $autoReg;
        this.regName = $regName;
        this.shader3D = $shader3D;

    }
}

File: listios/StorageStarted/pan3d/material/MaterialLoad.m
//
//  MaterialLoad.m
//  iosgl
//
//  Created by zhao on 9/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "MaterialLoad.h"
#import "Shader3D.h"
#import "Material.h"

@implementation MaterialLoad

- (instancetype)init:(SuccessMaterial)fun info:(NSDictionary*)info url:(NSString*)url autoReg:(BOOL)autoReg regName:(NSString*)regName shader:(Shader3D*)shader;
{
    self = [super init];
    if (self) {
        self.fun=fun;
        self.info=info;
        self.url=url;
        self.autoReg=autoReg;
        self.regName=regName;
        self.shader3D=shader;
    }
    return self;
}
//
//constructor($fun: Function, $info: any, $url: string, $autoReg: boolean, $regName: string, $shader3D: any) {
//       this.fun = $fun;
//       this.info = $info;
//       this.url = $url;
//
//       this.autoReg = $autoReg;
//       this.regName = $regName;
//       this.shader3D = $shader3D;
//   }
@end
